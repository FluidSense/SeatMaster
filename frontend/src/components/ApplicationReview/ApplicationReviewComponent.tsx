import { Column, Container, Row } from 'nav-frontend-grid';
import KnappBase from 'nav-frontend-knapper';
import PanelBase from 'nav-frontend-paneler';
import React, { Fragment } from 'react';
import { ILoginState } from '../Login/LoginReducer';
import { ACTION_NONE } from '../Login/Strings';
import {
  EDIT,
  EMAIL,
  NAME,
  NEEDS,
  NEW_APPLICATION,
  PARTNER,
  PHONE,
  PREFFERED_ROOM,
  SEAT_ROLLOVER,
} from './Strings';

const infoPanel = (title: string, text: string) => (
  <div>
    <h3><b>{title}</b></h3>
    <PanelBase>
      {text}
    </PanelBase>
  </div>
);

const newApplicationLink = (
  <KnappBase
    type="hoved"
  >
    {NEW_APPLICATION}
  </KnappBase>
);

interface IProps {
  applicationInfo: ILoginState;
}

const userInfo = () => (
    <Fragment>
      <Row className="10">
        {infoPanel(NAME, 'PÃ¥l Larsen')}
      </Row>
      <Row className="3">
        <Column lg="6">
          {infoPanel(EMAIL, 'paalar@kul.no')}
        </Column>
        <Column lg="6">
          {infoPanel(PHONE, '12345678')}
        </Column>
      </Row>
    </Fragment>
);

const roomInfo = () => (
  <Fragment>
    <Row className="2">
      <Column lg="4">
        {infoPanel(PARTNER, 'Peter Rydberg')}
      </Column>
      <Column lg="4">
        {infoPanel(PREFFERED_ROOM, 'Big one')}
      </Column>
      <Column lg="4">
        {infoPanel(SEAT_ROLLOVER, 'Requested')}
      </Column>
    </Row>
  </Fragment>
);

const ApplicationReviewComponent: React.FunctionComponent<IProps> = (props) => {
  const { status } = props.applicationInfo;
  if (status === ACTION_NONE) return newApplicationLink;
  return (
    <Fragment>
        {userInfo()}
        {roomInfo()}
        <Row className="">
          <Column lg="12">
            {infoPanel(NEEDS, 'Lorem ipsum schmoobedi boopLorem ipsum schmoobedi boopLorem ipsum schmoobedi boopLorem ipsum schmoobedi boopLorem ipsum schmoobedi boopLorem ipsum schmoobedi boopLorem ipsum schmoobedi boopLorem ipsum schmoobedi boopLorem ipsum bedi boopLorem ipsum schm schmoobedi boopLorem ipsum schmoobedi boop.')}
          </Column>
        </Row>
        <KnappBase
          type="hoved"
        >
          {EDIT}
        </KnappBase>
    </Fragment>
  );
};

export default ApplicationReviewComponent;
