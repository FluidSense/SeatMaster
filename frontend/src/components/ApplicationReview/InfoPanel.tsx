import PanelBase from 'nav-frontend-paneler';
import { Element } from 'nav-frontend-typografi';
import React from 'react';

interface IProps {
  title: string;
  text?: string;
}

const InfoPanel: React.FunctionComponent<IProps> = (props) => {
  const { title, text } = props;
  return (
    <div className={'info-panel'} id={`info-panel-${title.replace(' ', '')}`}>
      <Element>{title}</Element>
      <PanelBase border={true}>{text == null ? '' : text}</PanelBase>
    </div>
  );
};

export default InfoPanel;
