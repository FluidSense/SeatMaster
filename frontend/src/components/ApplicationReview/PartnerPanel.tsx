import AlertStripe, { AlertStripeInfo } from 'nav-frontend-alertstriper';
import { HjelpetekstVenstre } from 'nav-frontend-hjelpetekst';
import { LenkepanelBase } from 'nav-frontend-lenkepanel';
import PanelBase from 'nav-frontend-paneler';
import { Element } from 'nav-frontend-typografi';
import React from 'react';
import { NavLink } from 'react-router-dom';
import isEmpty from '../../utils/objectIsEmpty';
import { IApplication } from '../Application';
import { _CONNECTED_INFO, _PARTNER, _USER_NO_PARTNER } from './strings';

interface IProps {
  isAdmin?: boolean;
  partnerApplication?: IApplication;
}

const PartnerPanel: React.FunctionComponent<IProps> = (props) => {
  const { isAdmin, partnerApplication } = props;
  if (isAdmin) {
    if (partnerApplication && !isEmpty(partnerApplication)) {
      const url = `/admin/applications/${partnerApplication.id}/`;
      const link = (panelProps: any) => (
        <NavLink
            to={{ pathname: panelProps.href, state: {application: { ...partnerApplication } }}}
            className="lenkepanel lenkepanel--border"
        >
          {panelProps.children}
        </NavLink>);
      return (
        <div id="info-panel-Partner">
          <Element>{_PARTNER}</Element>
          <LenkepanelBase
            linkCreator={link}
            href={url}
          >
            {partnerApplication.user.fullname}
          </LenkepanelBase>
        </div>
      );
    }
    return (
      <div id="info-panel-Partner">
        <Element>{_PARTNER}</Element>
        <AlertStripeInfo>{_USER_NO_PARTNER}</AlertStripeInfo>
      </div>
    );
  }
  if (partnerApplication && !isEmpty(partnerApplication)) {
    return (
      <div id="info-panel-Partner">
        <Element>{_PARTNER}</Element>
        <AlertStripe type="suksess" solid={true}>{partnerApplication.user.fullname}</AlertStripe>
      </div>
    );
  }
  return (
      <div id="info-panel-Partner">
        <Element>{_PARTNER}</Element>
        <div className="panel-with-help-text">
          <PanelBase border={true} className="info-panel panel">{' '}</PanelBase>
          <HjelpetekstVenstre
            id="partner-help-text"
            tittel=""
          >
            {_CONNECTED_INFO}
          </HjelpetekstVenstre>
        </div>
      </div>
  );
};

export default PartnerPanel;
