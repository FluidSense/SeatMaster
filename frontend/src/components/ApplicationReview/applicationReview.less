#application-review {
  width: 80%;
}

#room-information {
  display: grid;
  grid-template-columns: 25% 25% 25% 25%;
  grid-template-rows: 100%;
}

#user-information {
  display: grid;
  grid-template-columns: 25% 25% 25% 25%;
  grid-template-rows: 50% 50%;
}

.info-panel {
  margin-bottom: 1rem;
  margin-right: 1rem;
}

#info-panel-Name {
  grid-column: 1 /span 2;
  grid-row: 1;
}

#info-panel-Masterstatus {
  grid-column: 3 /span 2;
  grid-row: 1;
}

#info-panel-E-Mail {
  grid-column: 1 /span 2;
  grid-row: 2;
}

#info-panel-Phone {
  grid-column: 3 /span 2;
  grid-row: 2;
}

#info-panel-Preferredroom {
  grid-column: 1 /span 2;
  grid-row: 1;
}

#info-panel-Partner {
  grid-column: 4;
  grid-row: 1;
}

#info-panel-Seatrollover {
  grid-column: 3;
  grid-row: 1;
}

#edit-application {
  margin-left: 0;
  margin-bottom: 1rem;
}
