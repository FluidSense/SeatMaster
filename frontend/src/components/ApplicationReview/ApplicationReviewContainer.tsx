import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import { IStore } from '../../store';
import { ILoginState } from '../Login/LoginReducer';
import ApplicationReviewComponent from './ApplicationReviewComponent';

interface IStateProps {
  applicationInfo: ILoginState;
}

type Props = IStateProps;

// tslint:disable-next-line:class-name
class _ApplicationReviewContainer extends React.Component<Props, {}> {
  public render = () => {
    const { applicationInfo } = this.props;
    return (
      <Fragment>
        <h1>Review Application</h1>
        <ApplicationReviewComponent
          applicationInfo={applicationInfo}
        />
      </Fragment>
    );
  }
}

const mapStateToProps = (state: IStore) => ({
  applicationInfo: state.userInformation,
});

const ApplicationReviewContainer = connect(
  mapStateToProps,
  null,
)(_ApplicationReviewContainer);

export default ApplicationReviewContainer;
