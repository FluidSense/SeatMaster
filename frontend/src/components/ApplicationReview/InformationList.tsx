import React from 'react';
import InfoPanel from './InfoPanel';
import { IRoomInfObject, IUserInfoObject } from './Presentational';
interface IProps {
  information: validProps;
}

type validProps = IRoomInfObject | IUserInfoObject;

const InformationList: React.FunctionComponent<IProps> = (props) => {
  const { information } = props;
  const list = Object.entries(information).map(array => (
    <InfoPanel key={array[0]} title={array[0]} text={array[1]} />));
  return (
    <>
      {list}
    </>
  );
};

export default InformationList;
