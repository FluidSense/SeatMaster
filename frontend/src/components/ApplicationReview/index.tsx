import React from 'react';
import { connect } from 'react-redux';
import { IStore } from '../../store';
import { IApplication } from '../Application';
import { IRegisteredUserState } from './../RegisterUser/reducer';
import './applicationReview.css';
import Presentational from './Presentational';

interface IStateProps {
  application?: IApplication;
  userInfo: IRegisteredUserState;
}

// tslint:disable-next-line:class-name
class _Container extends React.Component<IStateProps> {
  public render = () => {
    const { application, userInfo } = this.props;
    return (
      <Presentational application={application} userInfo={userInfo} />
    );
  }
}

const mapStateToProps = (state: IStore) => ({
  applicationInfo: state.applications.registeredApplication,
  userInfo: state.userInformation,
});

const ApplicationReview = connect(
  mapStateToProps,
  null,
)(_Container);

export default ApplicationReview;
