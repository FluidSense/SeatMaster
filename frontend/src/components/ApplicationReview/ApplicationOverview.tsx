import KnappBase from 'nav-frontend-knapper';
import { Sidetittel } from 'nav-frontend-typografi';
import React from 'react';
import { Link } from 'react-router-dom';
import { boolToString } from '../../utils/typeFormatter';
import { IApplication } from '../Application/index';
import InfoPanel from './InfoPanel';
import InformationList from './InformationList';
import PartnerPanel from './PartnerPanel';
import {
  _APPLICATION_INFO_TEXT,
  _COMMENTS,
  _EDIT_APPLICATION,
  _EMAIL,
  _MASTER_STATUS,
  _NAME,
  _NEEDS,
  _NEW_APPLICATION,
  _PARTNER,
  _PHONE,
  _PREFERRED_ROOM,
  _REVIEW_APPLICATION,
  _SEAT_ROLLOVER,
} from './strings';

interface IProps {
  application: IApplication;
  title?: string;
  pathToEdit?: string;
  isAdmin?: boolean;
}

export interface IInformationObject {
  [_COMMENTS]?: string;
  [_NEEDS]?: string;
}

export interface IUserInfoObject {
  [_NAME]?: string;
  [_EMAIL]?: string;
  [_PHONE]?: string;
  [_MASTER_STATUS]?: string;
}

export interface IRoomInfoObject {
  [_PREFERRED_ROOM]?: string;
  [_PARTNER]?: JSX.Element;
  [_SEAT_ROLLOVER]?: string;
}

const ApplicationOverview: React.FunctionComponent<IProps> = (props) => {
  const { application, title, isAdmin } = props;
  if (!application.user) return null;
  const userInfoObject: IUserInfoObject = {
    [_NAME]: application.user.fullname,
    [_EMAIL]: application.user.email,
    [_MASTER_STATUS]: application.rank,
  };

  const roomInfoObject: IRoomInfoObject = {
    [_PREFERRED_ROOM]: application.preferredRoom,
    [_PARTNER]: (
    <PartnerPanel
      key={125153}
      isAdmin={isAdmin}
      partnerApplication={application.partnerApplication}
    />),
    [_SEAT_ROLLOVER]: boolToString(application.seatRollover),
  };
  return (
    <div id="application-review">
      <Sidetittel>{title ? title : _REVIEW_APPLICATION}</Sidetittel>
      <p style={{ fontStyle: 'italic' }} >{_APPLICATION_INFO_TEXT}</p>
      <div id="user-information">{<InformationList information={userInfoObject} />}</div>
      <div id="room-information">{<InformationList information={roomInfoObject} />}</div>
      <div className={'needs-information'}>
        <InfoPanel title={_NEEDS} text={application.needs} />
        <InfoPanel title={_COMMENTS} text={application.comments} />
      </div>
      <Link to={props.pathToEdit ? props.pathToEdit : ''}>
        <KnappBase id="edit-application" type="hoved">{_EDIT_APPLICATION}</KnappBase>
      </Link>
    </div>
  );
};

export default ApplicationOverview;
