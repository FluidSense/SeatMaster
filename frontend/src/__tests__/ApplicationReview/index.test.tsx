import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import ApplicationReview from '../../components/ApplicationReview/index';

const mockStore = configureMockStore();
const applicationInfoMock = {
  email: 'test@test.com',
  fullname: 'Teser Testersen',
  partner: 'None',
  phone: '91100999',
  room: 'Space Commander',
  seatRollover: 'Yes',
  status: 'Master of masters',
};

describe('Application review index', () => {
  it('Renders correctly', () => {
    const initialState = { userInformation: applicationInfoMock };
    const store = mockStore(initialState);
    const wrapper = shallow(
      <Provider store={store}>
        <ApplicationReview />
      </Provider>,
    ).html();
    expect(wrapper).toMatchSnapshot();
  });
});
