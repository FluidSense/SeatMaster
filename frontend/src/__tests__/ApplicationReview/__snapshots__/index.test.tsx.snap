// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application review index Renders correctly 1`] = `"<h1 class=\\"typo-sidetittel\\">Review Application</h1><div id=\\"user-information\\"><div class=\\"info-panel\\" id=\\"info-panel-Name\\"><p class=\\"typo-element\\">Name</p><div class=\\"panel\\">Teser Testersen</div></div><div class=\\"info-panel\\" id=\\"info-panel-E-Mail\\"><p class=\\"typo-element\\">E-Mail</p><div class=\\"panel\\">test@test.com</div></div><div class=\\"info-panel\\" id=\\"info-panel-Phone\\"><p class=\\"typo-element\\">Phone</p><div class=\\"panel\\">91100999</div></div><div class=\\"info-panel\\" id=\\"info-panel-Masterstatus\\"><p class=\\"typo-element\\">Master status</p><div class=\\"panel\\">Master of masters</div></div></div><div id=\\"room-information\\"><div class=\\"info-panel\\" id=\\"info-panel-Partner\\"><p class=\\"typo-element\\">Partner</p><div class=\\"panel\\">None</div></div><div class=\\"info-panel\\" id=\\"info-panel-Preferredroom\\"><p class=\\"typo-element\\">Preferred room</p><div class=\\"panel\\">Space Commander</div></div><div class=\\"info-panel\\" id=\\"info-panel-SeatRollover\\"><p class=\\"typo-element\\">Seat Rollover</p><div class=\\"panel\\">Yes</div></div></div><div class=\\"needs-information\\"><div class=\\"info-panel\\" id=\\"info-panel-Needs\\"><p class=\\"typo-element\\">Needs</p><div class=\\"panel\\"></div></div></div><button class=\\"knapp knapp--hoved\\" id=\\"edit-application\\" type=\\"submit\\">Edit Application</button>"`;
